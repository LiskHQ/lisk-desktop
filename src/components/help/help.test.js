import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Help from './help';
import links from './../../constants/externalLinks';

describe('Help Page', () => {
  let wrapper;
  let props;
  let rendered;

  beforeEach(() => {
    props = {
      account: {
        address: '123L',
      },
      t: key => key,
      settingsUpdated: jest.fn(),
    };

    window.open = jest.fn();
    window.open.mockReturnValue({ focus: jest.fn() });
    wrapper = shallow(<Help {...props} />);
    rendered = renderer.create(<Help {...props} />).toJSON();
  });

  it('renders three help article sections with proper btn callbacks', () => {
    expect(rendered).toMatchSnapshot();

    wrapper.find('.help-onboarding').simulate('click');
    expect(props.settingsUpdated).toHaveBeenCalledWith({ onBoarding: true });

    wrapper.find('.help-visit-center').simulate('click');
    expect(window.open).toHaveBeenCalledWith(links.helpCenter, '_blank');

    const propsNotLoggedIn = {
      ...props,
    };
    delete propsNotLoggedIn.account.address;
    wrapper.setProps(propsNotLoggedIn);
    wrapper.update();
    expect(rendered).toMatchSnapshot();
  });
});
