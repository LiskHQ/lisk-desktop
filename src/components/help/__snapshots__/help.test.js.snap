// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help Page renders three help article sections with proper btn callbacks 1`] = `
<div
  className="wrapper wrapper"
>
  <aside
    className="col-sm-12 col-md-4 sideBar"
  >
    <header>
      <h2>
        Help
      </h2>
      <p>
        Browse through our tutorials, check out the FAQ or connect with our knowledgeable community.
      </p>
      <a
        className="help-onboarding"
        onClick={[Function]}
      >
        New to Hub? Take a tour
        <span
          className="icon arrow-right "
        />
      </a>
    </header>
  </aside>
  <section
    className="col-sm-12 col-md-8 helpSection help-articles"
  >
    <article>
      <h2>
        <span
          className="icon logo-icon headerIcon"
        />
        Help Center
      </h2>
      <div
        className="articleContainer"
      >
        <div>
          <p>
            Search for answers in our extensive 
            <span
              className="inlineIcon"
            >
              <span
                className="icon info "
              />
              <b>
                 FAQ
              </b>
            </span>
             or get in touch in via 
            <span
              className="inlineIcon"
            >
              <span
                className="icon conversation "
              />
              <b>
                 Chat
              </b>
            </span>
            .
          </p>
        </div>
        <div>
          <button
            className="help-visit-center"
            data-react-toolbox="button"
            disabled={false}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Visit Help Center
            <span
              className="icon arrow-right "
            />
          </button>
        </div>
      </div>
    </article>
    <article>
      <h2>
        <span
          className="icon academy headerIcon"
        />
        Academy
      </h2>
      <div
        className="articleContainer"
      >
        <div>
          <p>
            Learn about blockchain with our comprehensive knowledge base.
          </p>
        </div>
        <div>
          <ul>
            <li>
              <a
                href="https://lisk.io/academy/blockchain-basics/what-is-blockchain/eli5-blockchain"
                rel="noopener noreferrer"
                target="_blank"
              >
                Explain Blockchain Like I'm 5
                <span
                  className="icon arrow-right "
                />
              </a>
            </li>
            <li>
              <a
                href="https://lisk.io/academy/blockchain-basics/benefits-of-blockchain/blockchain-security"
                rel="noopener noreferrer"
                target="_blank"
              >
                Is Blockchain Secure?
                <span
                  className="icon arrow-right "
                />
              </a>
            </li>
            <li>
              <a
                href="https://lisk.io/academy/blockchain-basics/how-does-blockchain-work"
                rel="noopener noreferrer"
                target="_blank"
              >
                See more
                <span
                  className="icon arrow-right "
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </article>
    <article>
      <h2>
        <span
          className="icon lisk-chat headerIcon"
        />
        Lisk.Chat
      </h2>
      <div
        className="articleContainer"
      >
        <div>
          <p>
            Don’t be a stranger! Connect with our worldwide community.
          </p>
        </div>
        <div>
          <ul>
            <li>
              <a
                href="https://lisk.chat/home"
                rel="noopener noreferrer"
                target="_blank"
              >
                Visit Lisk.Chat
                <span
                  className="icon arrow-right "
                />
              </a>
            </li>
            <li>
              <a
                href="https://lisk.chat/channel/help-desk"
                rel="noopener noreferrer"
                target="_blank"
              >
                See #help-desk channel
                <span
                  className="icon arrow-right "
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </article>
  </section>
</div>
`;

exports[`Help Page renders three help article sections with proper btn callbacks 2`] = `
<div
  className="wrapper wrapper"
>
  <aside
    className="col-sm-12 col-md-4 sideBar"
  >
    <header>
      <h2>
        Help
      </h2>
      <p>
        Browse through our tutorials, check out the FAQ or connect with our knowledgeable community.
      </p>
      <a
        className="help-onboarding"
        onClick={[Function]}
      >
        New to Hub? Take a tour
        <span
          className="icon arrow-right "
        />
      </a>
    </header>
  </aside>
  <section
    className="col-sm-12 col-md-8 helpSection help-articles"
  >
    <article>
      <h2>
        <span
          className="icon logo-icon headerIcon"
        />
        Help Center
      </h2>
      <div
        className="articleContainer"
      >
        <div>
          <p>
            Search for answers in our extensive 
            <span
              className="inlineIcon"
            >
              <span
                className="icon info "
              />
              <b>
                 FAQ
              </b>
            </span>
             or get in touch in via 
            <span
              className="inlineIcon"
            >
              <span
                className="icon conversation "
              />
              <b>
                 Chat
              </b>
            </span>
            .
          </p>
        </div>
        <div>
          <button
            className="help-visit-center"
            data-react-toolbox="button"
            disabled={false}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Visit Help Center
            <span
              className="icon arrow-right "
            />
          </button>
        </div>
      </div>
    </article>
    <article>
      <h2>
        <span
          className="icon academy headerIcon"
        />
        Academy
      </h2>
      <div
        className="articleContainer"
      >
        <div>
          <p>
            Learn about blockchain with our comprehensive knowledge base.
          </p>
        </div>
        <div>
          <ul>
            <li>
              <a
                href="https://lisk.io/academy/blockchain-basics/what-is-blockchain/eli5-blockchain"
                rel="noopener noreferrer"
                target="_blank"
              >
                Explain Blockchain Like I'm 5
                <span
                  className="icon arrow-right "
                />
              </a>
            </li>
            <li>
              <a
                href="https://lisk.io/academy/blockchain-basics/benefits-of-blockchain/blockchain-security"
                rel="noopener noreferrer"
                target="_blank"
              >
                Is Blockchain Secure?
                <span
                  className="icon arrow-right "
                />
              </a>
            </li>
            <li>
              <a
                href="https://lisk.io/academy/blockchain-basics/how-does-blockchain-work"
                rel="noopener noreferrer"
                target="_blank"
              >
                See more
                <span
                  className="icon arrow-right "
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </article>
    <article>
      <h2>
        <span
          className="icon lisk-chat headerIcon"
        />
        Lisk.Chat
      </h2>
      <div
        className="articleContainer"
      >
        <div>
          <p>
            Don’t be a stranger! Connect with our worldwide community.
          </p>
        </div>
        <div>
          <ul>
            <li>
              <a
                href="https://lisk.chat/home"
                rel="noopener noreferrer"
                target="_blank"
              >
                Visit Lisk.Chat
                <span
                  className="icon arrow-right "
                />
              </a>
            </li>
            <li>
              <a
                href="https://lisk.chat/channel/help-desk"
                rel="noopener noreferrer"
                target="_blank"
              >
                See #help-desk channel
                <span
                  className="icon arrow-right "
                />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </article>
  </section>
</div>
`;
